src/arrayMethodJoin.js

```diff
@@ -4,9 +4,31 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = (',')) {
+    let result = '';
+
+    if (!this.length) {
+      return '';
+    }
```

Comment:
I'm convinced you can easily get rid of this block it does nothing

Reply:
Agree, fixed.